### PR TITLE
Refacto media library

### PR DESCRIPTION
**Pull request by ngilain**

### CODE REVIEW

File: en/developer_guide/media_gaufrette.rst [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
+
+If another Gaufrette filesystem is used instead of the default one, it should be notified to Open Orchestra through
+the ``open_orchestra_media.filesystem`` configuration parameter, which expects the identifier of the new filesystem
+to use.
 
-All adapters that can be natively used are described in the `configuration documentation`_,
-along with the details of the configuration to set up.
+Upload strategies
+-----------------
+When a new file is uploaded, a media document is created and the file is moved to the storage vie Gaufrette. But
```
Comment (aloupfor) [inline en/developer_guide/media_gaufrette.rst]: Should "vie" be "via" or "by"?

File: en/developer_guide/media_gaufrette.rst [hunk 2]
```diff
@@ -1,39 +1,52 @@
...
+some required process are also made whith files, and some other optional can be too. For instance, to be rendered
+in the Back Office gallery, a preview thumbnail must be generated. If the media is an image, several alternatives
+in different sizes must also be generated. In the file is a sound, maybe some new variants could be generated with
```
Comment (Thiblef) [inline en/developer_guide/media_gaufrette.rst]: If the file

File: en/developer_guide/media_gaufrette.rst [hunk 3]
```diff
@@ -1,39 +1,52 @@
...
+different bitrate. Those operations are completed by the alternatives strategies. A strategy exists for each
+supported mime-type.
 
-If another Gaufrette filesystem is used instead of the default one, it should be
-notified to Open Orchestra through the ``open_orchestra_media.filesystem`` configuration parameter,
-which expects the identifier of the new filesystem to use.
+When a new media is created due to a file upload, the ``MediaEvents::MEDIA_ADD`` event is fired. The
+``OpenOrchestra\MediaAdmin\EventSubscriber\MediaCreatedSubscriber`` catch this event to track the media created.
+Later after the response matching the media creation has been sent to the client, this subscriber also catch the
+``KernelEvents::TERMINATE`` event. It then ask the ``OpenOrchestra\MediaAdmin\FileAlternatives\FileAlternativesManager``
```
Comment (aloupfor) [inline en/developer_guide/media_gaufrette.rst]: Should "ask" be "asks"?

File: en/developer_guide/media_gaufrette.rst [hunk 4]
```diff
@@ -1,39 +1,52 @@
...
 
-All adapters that can be natively used are described in the `configuration documentation`_,
-along with the details of the configuration to set up.
+Upload strategies
+-----------------
+When a new file is uploaded, a media document is created and the file is moved to the storage vie Gaufrette. But
+some required process are also made whith files, and some other optional can be too. For instance, to be rendered
+in the Back Office gallery, a preview thumbnail must be generated. If the media is an image, several alternatives
+in different sizes must also be generated. In the file is a sound, maybe some new variants could be generated with
+different bitrate. Those operations are completed by the alternatives strategies. A strategy exists for each
```
Comment (Thiblef) [inline en/developer_guide/media_gaufrette.rst]: "bitrates"
"alternative strategies"

File: en/developer_guide/media_gaufrette.rst [hunk 5]
```diff
@@ -1,39 +1,52 @@
...
+If another Gaufrette filesystem is used instead of the default one, it should be notified to Open Orchestra through
+the ``open_orchestra_media.filesystem`` configuration parameter, which expects the identifier of the new filesystem
+to use.
 
-All adapters that can be natively used are described in the `configuration documentation`_,
-along with the details of the configuration to set up.
+Upload strategies
+-----------------
+When a new file is uploaded, a media document is created and the file is moved to the storage vie Gaufrette. But
+some required process are also made whith files, and some other optional can be too. For instance, to be rendered
```
Comment (aloupfor) [inline en/developer_guide/media_gaufrette.rst]: Should "whith" be "with"?

File: en/developer_guide/media_gaufrette.rst [hunk 6]
```diff
@@ -1,39 +1,52 @@
...
+in the Back Office gallery, a preview thumbnail must be generated. If the media is an image, several alternatives
+in different sizes must also be generated. In the file is a sound, maybe some new variants could be generated with
+different bitrate. Those operations are completed by the alternatives strategies. A strategy exists for each
+supported mime-type.
 
-If another Gaufrette filesystem is used instead of the default one, it should be
-notified to Open Orchestra through the ``open_orchestra_media.filesystem`` configuration parameter,
-which expects the identifier of the new filesystem to use.
+When a new media is created due to a file upload, the ``MediaEvents::MEDIA_ADD`` event is fired. The
+``OpenOrchestra\MediaAdmin\EventSubscriber\MediaCreatedSubscriber`` catch this event to track the media created.
```
Comment (Thiblef) [inline en/developer_guide/media_gaufrette.rst]: "the created media"

File: en/developer_guide/media_gaufrette.rst [hunk 7]
```diff
@@ -1,39 +1,52 @@
...
+Later after the response matching the media creation has been sent to the client, this subscriber also catch the
+``KernelEvents::TERMINATE`` event. It then ask the ``OpenOrchestra\MediaAdmin\FileAlternatives\FileAlternativesManager``
+to generate a thumbnail and the alternatives. So, the manager search a strategy able to deal with the uploaded file.
 
-Upload and resizing of images
------------------------------
+You can alter the alternatives generation by extending the existing strategies or by creating new ones.
```
Comment (Thiblef) [inline en/developer_guide/media_gaufrette.rst:37]: "alternative generations"

File: en/user_guide/media.rst [hunk 8]
```diff
@@ -1,67 +1,81 @@
-Media Management
-================
+The Media Library
+=================
 
-Open Orchestra allows you to manage medias uploaded by the users in the media library panel.
-They can be easily used in other part of the website such as blocks or contents.
-Natively, images, PDF and videos are supported by Open Orchestra.
+With the media bundles (MediaAdminBundle, MediaBundle and MediaFileBundle), Open Orchestra gets a consistant media
+library. This library allows you to upload media, manage them and reuse them on multiple locations of your site,
```
Comment (Thiblef) [inline en/user_guide/media.rst]: "to upload medias"

File: en/developer_guide/media_gaufrette.rst (hunk 7 above)
Comment (ngilain) [inline en/developer_guide/media_gaufrette.rst:37]: A least this one is not a fault :D

File: en/user_guide/media.rst [hunk 9]
```diff
@@ -1,67 +1,81 @@
...
+To remove a folder, first select the folder. The gallery displaying the content of the folder then appears. You can
+remove that folder only if it is empty. If this is the case, the toolbar located on the top right hand side of the
+gallery will contain a trash icon. Click on this icon to delete the folder.
 
 Manage files
 ------------
 
-The add button in the folder's content allows you to upload files.
+To add a file, select the folder you want to store it. Then on the bottom of the gallery, click the "add button".
+You can upload files in 3 different ways. The hole area is droppable, wich mean you can drag'n'drop files from your
```
Comment (Thiblef) [inline en/user_guide/media.rst]: "which means"

File: en/user_guide/media.rst [hunk 10]
```diff
@@ -1,67 +1,81 @@
...
 
-Once a format selected, the preview is updated and some choices are added:
+Image alternatives
+~~~~~~~~~~~~~~~~~~
 
-* manual crop: to adjust your selection manually
-* upload an alternative image: if user want to resize image himself, replace the selected thumbnail format by another image
+When you upload an image, several alternatives in different sizes are automatically generated to improve the
+rendering in the front office. According to the context the image could be displayed as a thumbnail or as a big
+poster. Out of the box, Open Orchestra generat three differents alternatives:
```
Comment (Thiblef) [inline en/user_guide/media.rst]: "different alternatives"

File: en/user_guide/media.rst [hunk 11]
```diff
@@ -1,67 +1,81 @@
...
+* Fixed height: Adapt height and keep proportion
+* Fixed width: Adapt width and keep proportion
+* Rectangle: Reduce to fit in the rectangle format, keeping the proportion
 
-Meta
-~~~~
+These ones can be personalized on a custom installation.
 
-Media meta information form have different fields :
+But you may want to alter an automatically generated alternative with an image more specific. You have two ways
```
Comment (Thiblef) [inline en/user_guide/media.rst]: "with a more specific image"

File: en/user_guide/media.rst [hunk 12]
```diff
@@ -1,67 +1,81 @@
...
+clicking on it.
 
-All files are organised inside folders. It's possible to browse the folder tree to see files by clicking on the navigation panel.
+Create a folder
+---------------
 
-Create
-~~~~~~
+To create a folder, browse the folders tree where you want to create the folder and click on the "new folder"
+button. The creation form then appears and require you to fill the following fields:
```
Comment (aloupfor) [inline en/user_guide/media.rst]: "requires"?

File: en/user_guide/media.rst [hunk 13]
```diff
@@ -1,67 +1,81 @@
...
-It is not possible to delete a folder containing some medias
+To remove a folder, first select the folder. The gallery displaying the content of the folder then appears. You can
+remove that folder only if it is empty. If this is the case, the toolbar located on the top right hand side of the
+gallery will contain a trash icon. Click on this icon to delete the folder.
 
 Manage files
 ------------
 
-The add button in the folder's content allows you to upload files.
+To add a file, select the folder you want to store it. Then on the bottom of the gallery, click the "add button".
```
Comment (aloupfor) [inline en/user_guide/media.rst]: I think a word is missing.
"select the folder where you want to store it"?

File: en/user_guide/media.rst (hunk 9 above)
Comment (aloupfor) [inline en/user_guide/media.rst]: "The whole area is droppable"?

File: en/user_guide/media.rst (hunk 10 above)
Comment (aloupfor) [inline en/user_guide/media.rst]: "generates"?

**Comment by ngilain** [comment]
@Thiblef , @aloupfor : update done

File: en/developer_guide/media_standalone_display.rst [hunk 14]
```diff
@@ -25,19 +25,19 @@ kernel each time you will request an image.
...
 
-A simple solution would be to install only the ``open-orchestra-media-bundle`` as a single
+A simple solution would be to install only the ``open-orchestra-media-file-bundle`` as a single
 ``Symfony`` application.
 
 Basic installation
 ~~~~~~~~~~~~~~~~~~
 
-The ``open-orchestra-media-bundle`` can run on it's own if you require it in the ``composer.json`` file :
+The ``open-orchestra-media-file-bundle`` can run on it's own if you require it in the ``composer.json`` file :
```
Comment (aloupfor) [inline en/developer_guide/media_standalone_display.rst]: "on its own"?

**Comment by SOLenG** [comment]
same thing as arthur to "its", else it's ok for me :+1:

**Comment by aloupfor** [comment]
:+1:

**Comment by Thiblef** [comment]
:+1:

**Pull request merged**